api/ryml.i

```diff
@@ -117,6 +119,11 @@ void parse_substr(c4::substr s, c4::yml::Tree *t)
     c4::yml::parse(s, t);
 }
 
+std::string emit_std_string(const c4::yml::Tree &t)
+{
+    return c4::yml::emitrs<std::string>(t);
+}
```

Comment:
I see, the include is dictated by this. But then what is the specific pro for std::string in here? Would it not be possible to return a plain char* ? I'd rather stay away from std containers here, as they bring a lot of secondary linker questions with them. But if the arguments are strong enough, I may convinced otherwise.

Reply:
Returning `char*` won't work because the lifetime of the allocation would be tied a local within the function.  SWIG has wrappers for taking a std::string and converting to a python `str`, which is why `std::string` was used here.

Is there an allocating container in the `c4` library you would prefer over `std::string` to use with `emitrs`?  There are several viable solutions to use here, but `std::string` seemed like the most robust option at first glance.

Reply:
[char* should be fine with some SWIG plumbing taken care of](http://www.swig.org/Doc2.0/SWIGDocumentation.html#Library_nn11). Swig also provides [cstring.i](http://www.swig.org/Doc2.0/SWIGDocumentation.html#Library_nn12) which may be of use.

Reply:
> char* should be fine with some SWIG plumbing taken care of. 

This would require a `c4::yml::emitrs` method that operates on raw malloc'd `char*`s.  I don't believe that exists, does it?  

I also don't think that there is a difference in terms of allocation behavior between `%newobject` return `char *` and returing `std::string`.  I agree with the [other thread](https://github.com/biojppm/rapidyaml/pull/102#discussion_r555996812) discussion about pre-allocated output buffers, but that is a totally different case to returning a newly allocated buffer.

Reply:
I've changed this.

Reply:
> This would require a c4::yml::emitrs method that operates on raw malloc'd char*s. I don't believe that exists, does it?

Indeed, and you figured correctly how to do it the same way. Kudos.

A clarification regarding the difference between char* and std::string. The question is not allocation behavior, but of linking requirements. If we stick to the c library then portability is much better. The complexity of sticking with malloc is not that much higher in terms of development. OTOH, exposing std::string in the public API (albeit implicitly) now brings into question std library link types, runtimes and so on. All of that just goes away by sticking to malloc(). In my view, that's a definite win.